core/xs_protocol.ml

```diff
@@ -74,14 +74,17 @@ let to_string = function
 end
 
 let rec split_string ?limit:(limit=(-1)) c s =
-  let i = try String.index s c with Not_found -> -1 in
-  let nlimit = if limit = -1 || limit = 0 then limit else limit - 1 in
-  if i = -1 || nlimit = 0 then
-    [ s ]
-  else
-    let a = String.sub s 0 i
-    and b = String.sub s (i + 1) (String.length s - i - 1) in
-    a :: (split_string ~limit: nlimit c b)
+  let len = String.length s in
```

Comment:
maybe guard this to check that limit >= -1 and raise Invalid_argument otherwise? wont terminate if its less I think

Reply:
Good plan.